### PR TITLE
Add abstract worker pool.

### PR DESCRIPTION
**Pull request by a-urth**
- use worker pool in query
- change config values for syncing

### CODE REVIEW

File: ledger.go [hunk 1]
```diff
@@ -802,7 +810,9 @@ func (l *Ledger) query() {
 				vote.block = &block
 			}
 			l.metrics.queryLatency.Time(f)
-		}(p)
+		}
+
+		l.queueWorkerPool.Queue(func() { f(p) })
```
Comment (hasyimibhar) [inline ledger.go]: CI caught a data race somewhere here.

Comment (a-urth) [reply ledger.go]: I assume parameter enclosure for embeded function works differently

**Comment by a-urth** [comment]
@hasyimibhar no race conditions now. test failed due to timeout though. locally i can't reproduce it, assumptions that it might because of worse performance on CI.

**Comment by rkeene** [comment]
@a-urth Can you rebase this ?

**Comment by a-urth** [comment]
@rkeene sure, i will

**Comment by rkeene** [comment]
I'm reviewing this now

File: worker_pool.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+				case <-wp.stop:
+					return
+				}
+			}
+		}()
+	}
+}
+
+func (wp *WorkerPool) Queue(job func()) {
+	wp.bus <- job
```
Comment (rkeene) [inline worker_pool.go]: Can we add a check here to do something if the WorkerPool has been closed ?   Maybe `panic` or something ?

File: worker_pool.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+func (wp *WorkerPool) Stop() {
```
Comment (rkeene) [inline worker_pool.go]: Should it be possible to call `wp.Start()` again after `wp.Stop()` has been called -- or should it just be an error to do so ?

Comment (a-urth) [reply worker_pool.go]: no restarts intended. worker pool lifecycle expected to be same as application using it.

File: worker_pool.go (hunk 2 above)
Comment (a-urth) [reply worker_pool.go]: i think i should close `bus` channel and then here will be panic because of that

File: worker_pool.go (hunk 3 above)
Comment (rkeene) [reply worker_pool.go]: Excellent, closing `wp.bus` should solve this as well

File: worker_pool.go (hunk 2 above)
Comment (rkeene) [reply worker_pool.go]: Perfect

**Comment by a-urth** [comment]
this one messed up, i'll recreate it

**Pull request closed without merging**